content/en/real_user_monitoring/session_replay/mobile/_index.md

```diff
@@ -18,26 +18,97 @@ Mobile Session Replay expands visibility into your mobile applications by visual
+
+The above screen can be reconstructed in 19 passes of the renderer:
+
+| Iteration | 1 | 2 | 3 |
+|-----------|---|---|---|
+| Viewport | {{< img src="real_user_monitoring/session_replay/mobile/how-it-works/how-it-works-iteration-1.png" alt="An example of a 'geometry' wireframe." style="width:100%;">}} | {{< img src="real_user_monitoring/session_replay/mobile/how-it-works/how-it-works-iteration-2.png" alt="An example of an 'image' wireframe." style="width:100%;">}} | {{< img src="real_user_monitoring/session_replay/mobile/how-it-works/how-it-works-iteration-3.png" alt="An example of a 'text' wireframe." style="width:100%;">}} |
+| Description | Draw "geometry" wireframe into viewport (rectangle + background color). | Draw "image" wireframe (rectangle + image with given UUID fetched from the Datadog backend). | Draw "text" wireframe (rectangle + "Twin Lake" text + font color, family and size). |
+
+The first wireframe could dictate the viewport size, enabling player size adjustment for given device + rotation (landscape / portrait).
+
+| Iteration | 4 | 5-9 | 10-12 |
+|-----------|---|---|---|
+| Viewport | {{< img src="real_user_monitoring/session_replay/mobile/how-it-works/how-it-works-iteration-4.png" alt="An example of a 'geometry', 'image, and 'text' wireframe." style="width:100%;">}} | {{< img src="real_user_monitoring/session_replay/mobile/how-it-works/how-it-works-iteration-5-9-1.png" alt="An example of a 'geometry' and 'image' wireframe." style="width:100%;">}} | {{< img src="real_user_monitoring/session_replay/mobile/how-it-works/how-it-works-iteration-10-12.png" alt="An example of a 'geometry' and 'image' wireframe." style="width:100%;">}} |
+| Description | Draw "image" and "text" wireframes into viewport.| Draw "image" and "text" wireframes into viewport. | Draw "image" and two "geometry" wireframes into viewport.  |
+
+Because wireframes are sorted in back-to-front order, it overdraws existing portion of the frame. This is very much desirable. For semi-transparent elements, it can also do the job of proper blending "closer" elements with the "farer" ones.
+
+| Iteration | 13-19 | Final result |
+|-----------|-------|--------------|
+| Viewport | {{< img src="real_user_monitoring/session_replay/mobile/how-it-works/how-it-works-iteration-13-19.png" alt="An example of a 'geometry' and 'image' wireframe." style="width:50%;">}} | {{< img src="real_user_monitoring/session_replay/mobile/how-it-works/how-it-works-iteration-final-result.png" alt="An example of a 'geometry' and 'image' wireframe." style="width:50%;">}} |
+
+### Full and incremental snapshots
+
+The sequence of all visible wireframes can be considered a full snapshot record if it is referring to the web format. Sending it per frame in the replay is not optimal, which is why the recorder takes incremental snapshots.
+
+To send incremental changes, the mobile SDK tags each wireframe with a unique identifier. Native views are objects, sot hey can be identified by reference.
```

Comment:
```suggestion
To tie Mobile Session Replay back to its [Browser counterpart][1], the sequence of all visible wireframes is considered a full snapshot. But to avoid re-recording the entire screen when only incremental changes occur in the UI, the mobile SDK tags each wireframe with a unique identifier, and sends incremental snapshots to Datadog. Since native views are objects, they can be identified by reference.
```